scripts/generate_manifest.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+import itertools
+from openpyxl import load_workbook
+import os
+
+from EPPs.common import SendMailEPP
+
+
+class GenerateManifest(SendMailEPP):
+    # populate the sample manifest with the sample date. Sample manifest template is determined by a step udf.
+    # The starting row and columns are determined by step UDFs. Uses SendMailEPP object for get_config function
+
+    _use_load_config = True  # should the config file be loaded?
+
+    # additional argument required to obtain the file location for newly create manifest in the LIMS step
+    def __init__(self, argv=None):
+        super().__init__(argv)
+        self.manifest = self.cmd_args.manifest
+
+    @staticmethod
+    def add_args(argparser):
+        argparser.add_argument(
+            '-m', '--manifest', type=str, required=True, help='Sample manifest generated by the LIMS'
+        )
+
+    def _run(self):
+
+        # obtain all of the inputs for the step
+        all_inputs = self.process.all_inputs(unique=True)
+        input_project_name=all_inputs[0].samples[0].project.name
+
+
+
+        # check all input containers have the same type and also create list of unique containers
+        container_types = set()
+        unique_container_names=set()
+
+
+        for artifact in all_inputs:
```

Comment:
There is now a property called `input_container_names` in common that you can  use instead. it is a cached_property so you calling it multiple time won't be expensive.

Reply:
Replaced use of "unique_container_names" with "input_container_names"